python-frontend/src/main/java/org/sonar/plugins/python/api/tree/ReturnStatement.java

```diff
@@ -36,4 +38,22 @@ public interface ReturnStatement extends Statement {
 
   @Beta
   List<Token> commas();
+
+  /**
+   * Infers the type of the returned value that would result from the execution of this return statement.
+   */
+  @Beta
+  default InferredType type() {
```

Comment:
To differentiate with `Expression#type()` and to make explicit this is about the return value, I would maybe call it `returnedType` or `returnValueType`